internal/controllers/clusterrequest_controller_test.go

```diff
@@ -3634,3 +3629,290 @@ var _ = Describe("hasPolicyConfigurationTimedOut", func() {
 		Expect(CRTask.object.Status.ClusterDetails.NonCompliantAt).ToNot(BeZero())
 	})
 })
+
+var _ = Describe("ZTPDone", func() {
+	var (
+		c                client.Client
+		ctx              context.Context
+		CRReconciler     *ClusterRequestReconciler
+		crName           = "cluster-1"
+		ctName           = "clustertemplate-a-v1"
+		ctNamespace      = "clustertemplate-a-v4-16"
+		ciDefaultsCm     = "clusterinstance-defaults-v1"
+		hwTemplateCm     = "hwTemplate-v1"
+		policy           *policiesv1.Policy
+		ptDefaultsCm     = "policytemplate-defaults-v1"
+		clusterRequestCR *oranv1alpha1.ClusterRequest
+		req              reconcile.Request
+	)
+
+	BeforeEach(func() {
+		// Define the needed resources.
```

Comment:
I am thinking could these tests be intergrated/injected into the e2e cases in the ClusterRequestReconcile? So we don't have to re-declare all the resources.